src/main/java/com/hierynomus/smbj/connection/Connection.java

```diff
@@ -217,6 +243,40 @@ public void handle(SMB2Packet packet) throws TransportException {
             // TODO reauthenticate session!
             return;
         }
+        
+        if (packet.getHeader().getSessionId() != 0 && (packet.getHeader().getMessage() != SMB2MessageCommandCode.SMB2_SESSION_SETUP)) {
+            Session session = connectionInfo.getSessionTable().find(packet.getHeader().getSessionId());
+            if (session == null) {
+                // check for a not-yet-authenticated session
+                session = connectionInfo.getPreauthSessionTable().find(packet.getHeader().getSessionId());
+                if (session == null) {
+                    logger.warn("Illegal request, no session matching the sessionId: {}", packet.getHeader().getSessionId());
+                    //TODO: maybe tear down the connection?
+                    return;
+                }
+            }
+            
+            // check packet signature.  Drop the packet if it is not correct.
+            if (session.getSigningRequired()) {
+                if (packet.getHeader().isFlagSet(SMB2MessageFlag.SMB2_FLAGS_SIGNED)) {
+                    packet.getBuffer().rpos(0);
+                    if (!MessageSigning.validateSignature(packet.getBuffer().array(), packet.getBuffer().available(), session.getSigningKey())) {
+                        logger.warn("Invalid packet signature");
```

Comment:
Do you mean the tunable parameter should be a field in the Config class?  And is it your intention to have a bunch of these kinds of items?  If so, let me know what kind of architectural style you'd like to use.  For instance, it could be a boolean method (like the other Config items) or it could be a System.getProperty() kind of thing.